### PR TITLE
VACMS-18544: next build feature flags

### PR DESCRIPTION
**Pull request by timcosgrove**
# Description
*Do not merge until https://github.com/department-of-veterans-affairs/va.gov-cms/pull/18937 has been merged the work from that PR has filtered down to the main lower Tugboat instance.*

This PR, in conjunction with [this PR in the CMS](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/18937), sets up Next Build to use CMS feature flags to determine whether a given content type should be built in Next Build. 

Please see https://github.com/department-of-veterans-affairs/next-build/pull/676/files?short_path=a217030#diff-a21703051db1f4ae84b593407423d494143eab7fe3f1f5db420ae1fb3935dad7 for how the feature flags are managed and used in Next Build.

## Ticket
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18544

## Developer Task

```[tasklist]
- [x] PR submitted against the `main` branch of `next-build`.
- [x] Link to the issue that this PR addresses (if applicable).
- [x] Define all changes in your PR and note any changes that could potentially be breaking changes.
- [x] PR includes steps to test your changes and links to these changes in the Tugboat preview (if applicable).
- [x] Provided before and after screenshots of your changes (if applicable).
- [x] Alerted the #accelerated-publishing Slack channel to request a PR review.
- [x] You understand that once approved, you are responsible for merging your changes into `main`. (Note that changes to `main` will move automatically into production.)
```

## Testing Steps
This is most easily tested with locally-running Next Build against a locally running CMS instance.
### Setup
1. In your local CMS instance, check out main, pull the latest DB (`ddev pull va --skip-files`), and then ensure it is running correctly. 
2. Check this branch out for Next Build.
3. Make sure that your .env.local file is pointed towards your local CMS in .env.local. Make sure this is `http://va-gov-cms.ddev.site` rather than `https`
4. Start Next Build locally: `yarn dev`

### First tests - enabling content types from the CMS.
1. Visit an events listing page: http://localhost:3999/pittsburgh-health-care/events/. This should return a 404/page not found.
2. Visit the system feature toggle page, https://va-gov-cms.ddev.site/admin/config/system/feature_toggle. Find the Event Listing flag, check it, and save (`FEATURE_NEXT_BUILD_CONTENT_EVENT_LISTING`). 
3. Stop and restart local Next Build.
4. Reload http://localhost:3999/pittsburgh-health-care/events/ (it may reload on its own). This should now load. 
5. Visit any event page, i.e. http://localhost:3999/pittsburgh-health-care/events/72337/. It should return a 404.
6. Visit the system feature toggle page, https://va-gov-cms.ddev.site/admin/config/system/feature_toggle. Find the Event  flag, check it, and save (`FEATURE_NEXT_BUILD_CONTENT_EVENT`).
7. Stop and restart local Next Build. 
8. Reload http://localhost:3999/pittsburgh-health-care/events/72337/ (it may reload on its own). This page should now load.  

### Enabling content types from the environment file.
1. Start your local Next Build if it is not running.
2. Visit a Vet Center page: http://localhost:3999/oklahoma-city-vet-center/. It should return a 404.
3. In your envs/.env.local file, add the following variable: `FEATURE_NEXT_BUILD_CONTENT_VET_CENTER=true`
4. Stop and restart local Next Build.
5. Reload http://localhost:3999/oklahoma-city-vet-center/ (it may reload on its own). The page should now load. 

### Testing the special all content variable.
1. Start your local Next Build if it is not running.
2. Visit a Story page: http://localhost:3999/portland-health-care/stories/portland-veteran-receives-first-place-award-for-poetry-entry/. It should return a 404.
3. In your envs/.env.local file, add the following variable: `FEATURE_NEXT_BUILD_CONTENT_ALL=true`
4. Stop and restart local Next Build.
5. Reload http://localhost:3999/portland-health-care/stories/portland-veteran-receives-first-place-award-for-poetry-entry/ (it may reload on its own). The page should now load. 

# Reviewer

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository. 

## Standard Checks

```[tasklist]
- [x] Code Quality: Readabilty, Naming Conventions, Consistency, Reusability
- [x] Test Coverage: 80% coverage
- [x] Functionality: Change functions as expected with no additional bugs
- [x] Performance: Code does not introduce performance issues
- [ ] Documentation: Changes are documented in their respective README.md files
- [x] Security: Packages have been approved in the TRM
```

### CODE REVIEW

File: packages/env-loader/src/index.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ export const processEnv = async (command: string): Promise<void> => {
   process.env = {
     ...process.env,
     ...{
-      ...envVars,
```
Comment (timcosgrove) [inline packages/env-loader/src/index.ts:36]: This ordering is switched to allow developer override of feature flags via their env vars, either locally or on a given environment. In the previous setup, the only way to override the feature flags was via cli options, and that will be difficult for devs to work with and nearly impossible to utilize on say a dev or staging environment.

**Review by nfpappas-oddball — DISMISSED** [review]
If not already done, please also link the feature flags docs inside of cms readme's so that no information is lost.

**Comment by nfpappas-oddball** [comment]
Everything seems to work great. Even when i check all options or even add some of the documented but not present options such as FEATURE_NEXT_BUILD_CONTENT_ALL. Everything worked as expected.

**Comment by ttran-Therisa** [comment]
The testing works for me. I also think the testing instructions would be a great to add to the readme as an example to serve as practice for the user

**Review by ttran-Therisa — APPROVED** [review]
lgtm

**Review by nfpappas-oddball — APPROVED** [review]

**Pull request merged**